packages/cli/lib/services/model.service.ts

```diff
@@ -1,211 +1,167 @@
-
-                if (hasStringLiteral) {
-                    const enumValues = tsType
-                        .split('|')
-                        .map((e) => (isJsOrTsType(e.trim()) || Object.keys(models).includes(e.trim()) ? e.trim() : `'${e.trim()}'`))
-                        .join(' | ');
-                    tsType = enumValues;
-                }
-            }
-
-            return tsType;
-        } else {
-            try {
-                const nestedFields = Object.keys(rawField)
-                    .map((fieldName: string) => `  ${fieldName}: ${this.getFieldType(rawField[fieldName] as string | NangoModel, debug, modelName, models)};`)
-                    .join('\n');
-                return `{\n${nestedFields}\n}`;
-            } catch (_) {
-                // eslint-disable-next-line no-console
-                console.log(chalk.red(`Failed to parse field ${JSON.stringify(rawField)} so just returning it back as a string`));
-                return String(rawField);
-            }
-        }
+    const modelTs = buildModelsTS({ parsed });
+    fs.writeFileSync(fp, modelTs);
+
+    if (debug) {
+        printDebug(`${TYPES_FILE_NAME} generated`);
     }
 
-    public async createModelFile({ fullPath, notify = false }: { fullPath: string; notify?: boolean }) {
-        const configContents = fs.readFileSync(path.join(fullPath, nangoConfigFile), 'utf8');
-        const configData: NangoConfig = yaml.load(configContents) as NangoConfig;
-        const { models, integrations } = configData;
-        const interfaceDefinitions = modelService.build(models, integrations);
-        if (interfaceDefinitions) {
-            fs.writeFileSync(path.join(fullPath, TYPES_FILE_NAME), interfaceDefinitions.join('\n'));
-        }
+    return { success: true, error: null, response: parsed };
+}
 
-        if (!fs.existsSync(`${getNangoRootPath()}/${NangoSyncTypesFileLocation}`)) {
-            throw new Error(`Failed to load ${NangoSyncTypesFileLocation}`);
-        }
+/**
+ * Build models.ts
+ */
+export function buildModelsTS({ parsed }: { parsed: NangoYamlParsed }): string {
+    const output: string[] = [
+        '// ---------------------------',
+        `// This file was generated by Nango (v${NANGO_VERSION})`,
+        "// It's recommended to version this file",
+        '// https://nango.dev',
+        '// ---------------------------',
+        ''
+    ];
+
+    output.push('// ------ Models');
+    output.push(generateInterfaces({ parsed }).join('\n\n'));
+    output.push('// ------ /Models');
+    output.push('');
+
+    output.push('// ------ SDK');
+    output.push(generateSDKTypes());
+    output.push('// ------ /SDK');
+    output.push('');
+
+    output.push(`// ------ Flows
+export const NangoFlows = ${JSON.stringify(parsed.integrations, null, 2)} as const;
+// ------ /Flows
+`);
```

Comment:
can you directly interpolate the output string to remove the noise?
```suggestion
return `
// ---------------------------
// This file was generated by Nango (v${NANGO_VERSION})
// It's recommended to version this file
// https://nango.dev
// ---------------------------

// ------ Models
${generateInterfaces({ parsed }).join('\n\n')}
// ------ /Models

// ------ SDK
${generateSDKTypes()}
// ------ /SDK

// ------ Flows
export const NangoFlows = ${JSON.stringify(parsed.integrations, null, 2)} as const;
// ------ /Flows
`;
```

Reply:
yeah true, it didn't was that simple at the beginning makes sense now 👍🏻